packages/common/src/hooks/useQueryParams/useQueryParams.ts

```diff
@@ -1,31 +1,50 @@
 import { ObjectWithStringKeys } from './../../types/utility';
 import { usePagination, PaginationState } from '../usePagination';
 import { useSortBy, SortState, SortRule } from '../useSortBy';
+import { useFilterBy, FilterState, FilterBy } from '../useFilterBy';
 
 export interface QueryParams<T extends ObjectWithStringKeys>
   extends SortState<T>,
+    FilterState<T>,
     PaginationState {}
 
 export interface QueryParamsState<T extends ObjectWithStringKeys>
   extends SortState<T>,
+    FilterState<T>,
     PaginationState {
   pagination: PaginationState;
+  sort: SortState<T>;
+  filter: FilterState<T>;
   queryParams: QueryParams<T>;
 }
 
-export const useQueryParams = <T extends ObjectWithStringKeys>(
-  initialSortBy: SortRule<T>
-): QueryParamsState<T> => {
+export const useQueryParams = <T extends ObjectWithStringKeys>({
+  initialSortBy,
+  initialFilterBy,
+}: {
+  initialSortBy: SortRule<T>;
+  initialFilterBy?: FilterBy<T>;
+}): QueryParamsState<T> => {
+  const filter = useFilterBy(initialFilterBy);
+  const sort = useSortBy(initialSortBy);
   const pagination = usePagination();
-  const { sortBy, onChangeSortBy } = useSortBy<T>(initialSortBy);
 
-  const queryParams = { ...pagination, pagination, sortBy, onChangeSortBy };
+  const queryParams: QueryParams<T> = {
+    ...pagination,
+    ...filter,
+    ...sort,
+    filter,
+    sort,
+    pagination,
+  };
 
   return {
     ...pagination,
+    ...sort,
+    ...filter,
+    sort,
+    filter,
```

Comment:
adding `useFilterBy` to `useQueryParams` (and also returning objects `sort` and `filter` with the respective states - eaiser to pass around 🤷